### DISCUSSION TITLE
List of tasks

### DISCUSSION THREAD

**Discussion by amorygalili** [frc2423/2024#56] [Week 2 competition]
This is a list of the tasks that we need done for our week 2 competition:

- [x] Fix bug where we're only setting one side of the shooter wheels when shooting from a distance. 
- [x] Add command to automatically move to spot on field while avoid obstacles: https://pathplanner.dev/pplib-pathfinding.html
- [x] Add command to rotate to target pose
- [x] Make sure all commands involving robot movement work for both red+blue alliances
- [x] Be able to disable shooting at distance with vision from dashboard. If disabled shooting at distance should use fixed distances. Maybe add buttons on dashboard to manually set distance?
- [x] Create pathplanner named auto commands to:
  1. perform intake/shooter handoff
  2. set shooter rev speed + angle based on distance from speaker
  3. shoot (run green wheels)
- [x] Make sure vision pose estimation is working while disabled so the robot knows where it is when autonomous starts
- [x] Update function to get distance from speaker so it works on both red+blue alliances
- [ ] Update 2 game piece autos to work with shoot at distance
- [ ] Update 3 game piece autos to work with shoot at distance
- [ ] Create 4 game piece shoot at distance auto
- [ ] Create center line autos
- [x] Be able to see current pathplanner trajectory in dashboard
- [x] Test shooter angle safety
- [ ] Use multiple beam break sensors on intake to improve reliability of game piece detection

**Comment by briankilcullen** [comment]
Test shooting from the podium distance+angle (not sure if you did this at WPI, but further testing+practice would not hurt even if you did and it worked).

**Comment by briankilcullen** [comment]
Check amp to make sure it still works, and practice. Amp -> amp -> shooter cycle sets are worth more than just doing the speaker even if none of our partners are sharing the amplifications and we can only score one amplified note ourselves, which would be the worst case (1 + 1 + 5 > 2 + 2 + 2).
